src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs

```diff
@@ -742,35 +743,163 @@ public override ClaimsPrincipal ValidateToken(string token, TokenValidationParam
             if (tokenParts.Length != JwtConstants.JwsSegmentCount && tokenParts.Length != JwtConstants.JweSegmentCount)
                 throw LogHelper.LogExceptionMessage(new ArgumentException(LogHelper.FormatInvariant(LogMessages.IDX12741, token)));
 
-            ClaimsPrincipal claimsPrincipal = null;
-            SecurityToken signatureValidatedToken = null;
-
+            ClaimsPrincipal claimsPrincipal;
+            SecurityToken signatureValidatedToken;
```

Comment:
if we were to pass 'validatedToken' to ValidateToken(...), we could remove this local variable and the assignment below.

Reply:
It will break some tests if write like this. For some tests if ValidteToken() throws an exception, this implementation will still out a token, and it shouldn't be happen.

Reply:
We realized we can set 'validatedToken' to null in the JWE method that successfully decrypts the JWE but fails the inner JWS validation.